flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartitionView.java

```diff
@@ -70,14 +75,30 @@ public boolean isReleased() {
 		return isReleased.get() || parent.isReleased();
 	}
 
+	@Override
+	public boolean isBlockedByCheckpoint() {
+		return isBlockedByCheckpoint;
+	}
+
+	@Override
+	public void resumeConsumption() {
+		checkState(isBlockedByCheckpoint, "Should be blocked by checkpoint.");
+
+		isBlockedByCheckpoint = false;
```

Comment:
we need to check the condition here and `notifyDataAvailable` if possible, otherwise it might cause that the partition view would never add into handler if there are already accumulated more than two buffers.